### PR TITLE
JNI Rolling Aggregation Changes [skip ci]

### PR DESCRIPTION
**Pull request by revans2**
This depends on #8052 

It adds in similar changes to the java APIs to have the compiler be able to check if an aggregation will work for rolling window or not.

### CODE REVIEW

File: java/src/main/java/ai/rapids/cudf/AggregationOverWindow.java [hunk 1]
```diff
@@ -22,10 +22,11 @@
  * An Aggregation instance that also holds a column number and window metadata so the aggregation
```
Comment (jlowe) [inline java/src/main/java/ai/rapids/cudf/AggregationOverWindow.java:22]: 2021 copyrights

File: java/src/main/java/ai/rapids/cudf/AggregationOnColumn.java [hunk 2]
```diff
@@ -22,18 +22,18 @@
  * An Aggregation instance that also holds a column number so the aggregation can be done on
```
Comment (jlowe) [inline java/src/main/java/ai/rapids/cudf/AggregationOnColumn.java:22]: 2021 copyrights

**Review by jlowe — APPROVED** [review]
So glad to see the aggregation `clone()` usage almost completely removed.  Would be great to get rid of that last one in `ColumnView.reduce` at some point.

**Comment by revans2** [comment]
@gpucibot merge

**Pull request merged**